misago/users/djangoadmin.py

```diff
@@ -1,96 +1,33 @@
-"""
+# -*- coding: utf-8 -*-
 from django.contrib import admin
-from django.contrib.auth import get_user_model
 from django.core.urlresolvers import reverse
 from django.utils.html import format_html
 from django.utils.translation import ugettext as _
 
-from misago.core import forms
-
-
-# Misago user model
-User = get_user_model()
-
-
-class UserAdminForm(forms.ModelForm):
-    """
-    This form adds `edit_from_misago_link` pseudo-field, that renders 
-    itself like an html hyperlink to a user edit page in misago admin panel.
-
-    This could be done with `User` edit template overwrite, but
-    it is kind of overkill - overwrite the whole template just to add one 
-    button - isn't it?
-    """
-    #: pseudo-field
-    edit_from_misago_link = forms.Field()
-    
-    def __init__(self, *args, **kwargs):
-        # noinspection PyArgumentList
-        super(UserAdminForm, self).__init__(*args, **kwargs)
-        self.init_edit_from_misago_link_field()
 
-    def init_edit_from_misago_link_field(self):
-        """
-        Init for the pseudo-field, and replace it's widget `render`.
-        """
-        field = self.fields['edit_from_misago_link']
-        field.required = False
-        field.label = ''
-        field.widget.render = self.render_edit_from_misago_link
-
-    # noinspection PyUnusedLocal
-    def render_edit_from_misago_link(self, *args, **kwargs):
-        """
-        Composes an html hyperlink for the pseudo-field render.
-
-        `*args` and `**kwargs` arguments need to mimic widget `render()`
-        signature.
-
-        :rtype: str
-        """
-        text = _('Edit this user in Misago admin panel')
-        link_html_template = ('<a href="{}" target="blank">' + text + '</a>')
-        link_url = reverse(
-            viewname='misago:admin:users:accounts:edit',
-            kwargs={'pk': self.instance.pk},
-        )
-        link_html = format_html(link_html_template, link_url)
-
-        return link_html
-    
-    class Meta:
-        model = User
-        fields = ['edit_from_misago_link']
+def _change_field_short_description(short_desc_new):
+    def func_wrapper(func):
+        func.short_description = short_desc_new
+        return func
+    return func_wrapper
 
 
 class UserAdminModel(admin.ModelAdmin):
     """
-    Redeclare most of the model fields like read-only. 
-    Prevents new/delete actions (users should use misago admin panel for 
-    that).
-    Replaces default form with custom `UserAdminForm`.
+    Removes new and delete actions, use misago admin panel for that.
+    The registration call is placed in misago.users.admin.
     """
-
     list_display = (
         'username',
         'email',
         'is_staff',
         'is_superuser',
+        'rank',
+        'show_edit_from_misago_url',
     )
     search_fields = ('username', 'email')
-    list_filter = ('groups', 'is_staff', 'is_superuser')
+    list_filter = ('groups', 'is_staff', 'is_superuser', 'rank')
```

Comment:
Lets also make rank readonly, there's some extra logic involved in rank change (eg. regeneration of ACL token) that django admin's unlikely to hangle.